syntax = "proto3";

package protos;

service FileExchanger {
  rpc ListFiles (Request) returns (Response) {}
  rpc DownloadFile (DownloadRequest) returns (stream FileChunkResponse) {}
}

message Request {
  string path = 1;
  // Deprecated: use new endpoint instead.
  bool show_hidden_files = 2;
}

message DownloadRequest {
  oneof path {
    string absolute_path = 1;
    string file_name = 2;
  }
}

message Response {
  message File {
    string name = 1;
    string size = 2;
    string path = 3;
  }
  string path = 1;
  repeated File files = 2;
}

message FileChunkResponse {
  bytes chunk = 1;
}

